Spigot-API-Patches/0236-Added-PlayerChangeBeaconEffectEvent.patch

```diff
@@ -0,0 +1,122 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Jake Potrebic <jake.m.potrebic@gmail.com>
+Date: Wed, 24 Jun 2020 15:12:18 -0600
+Subject: [PATCH] Added PlayerChangeBeaconEffectEvent
+
+
+diff --git a/src/main/java/io/papermc/paper/event/player/PlayerChangeBeaconEffectEvent.java b/src/main/java/io/papermc/paper/event/player/PlayerChangeBeaconEffectEvent.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..d662b99cebb5aa0b2e21c6ba0a076517fa2f48b1
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/event/player/PlayerChangeBeaconEffectEvent.java
+@@ -0,0 +1,110 @@
++package io.papermc.paper.event.player;
++
++import org.bukkit.block.Block;
++import org.bukkit.entity.Player;
++import org.bukkit.event.Cancellable;
++import org.bukkit.event.HandlerList;
++import org.bukkit.event.player.PlayerEvent;
++import org.bukkit.potion.PotionEffectType;
++import org.jetbrains.annotations.NotNull;
++import org.jetbrains.annotations.Nullable;
++
++/**
++ * Called when a player sets the effect for a beacon
++ */
++public class PlayerChangeBeaconEffectEvent extends PlayerEvent implements Cancellable {
++    private static final HandlerList handlers = new HandlerList();
++
++    private PotionEffectType primary;
++    private PotionEffectType secondary;
++    private final Block beacon;
++
++    private boolean isCancelled;
++
++    public PlayerChangeBeaconEffectEvent(@NotNull Player player, @Nullable PotionEffectType primary, @Nullable PotionEffectType secondary, @Nullable Block beacon) {
++        super(player); this.primary = primary; this.secondary = secondary; this.isCancelled = false; this.beacon = beacon;
++    }
++
++    /**
++     * @return the primary effect
++     */
++    @Nullable public PotionEffectType getPrimary() {
++        return primary;
++    }
++
++    /**
++     * Sets the primary effect
++     * <p>
++     * NOTE: The primary effect still has to be one of the valid effects for a beacon.
```

Comment:
Have you looked into whether it'd be possible to alleviate/remove this requirement?

Reply:
Yeah, it might be possible. Right now, TileEntityBeacon.a just has a final 2d array of possible effects for beacons, and if its not one of those, it just sets the effect to null or no effect. So maybe that's something that can be looked at in the future, but I don't think its in the scope of this PR.